### DISCUSSION TITLE
What happen if there is no hardware acceleration?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#202] [General]
ecode and all eepp applications will be useless (unable to run)? Or is there a fallback mode?

**Comment by SpartanJ** [comment]
That technically depends on each operating system, so:
Linux: it will always fallback to software rendering ([LLVMpipe](https://docs.mesa3d.org/drivers/llvmpipe.html)).
macOS: it will always fallback to software rendering.
Windows: depends on the Windows version. Windows 10+ will fallback to software rendering, Windows 7 and 8 should almost always fallback to software rendering but depends on the installation, it's possible to install also a software renderer based on LLVMpipe (a [MESA3D](https://github.com/pal1000/mesa-dist-win) distribution).
Haiku: it uses a LLVMPipe (OS doesn't provide hardware acceleration for the moment)

Having said that, the only common real world scenario where you can find the case of no hardware acceleration is when using virtualization.
